src/ngraph/runtime/gpu/gpu_external_function.cpp

```diff
@@ -635,28 +629,41 @@ void runtime::gpu::GPU_ExternalFunction::compile()
     {
         return;
     }
+    std::unique_lock<std::mutex> lock(s_compilation);
 
     m_function_name = m_function->get_name();
-    string dump_filename = file_util::path_join(s_output_dir, m_function_name + "_ops.txt");
 
-    // For now, just make everyone row-major.
-    m_pass_manager.register_pass<pass::ResultCopyElimination>();
-    m_pass_manager.register_pass<pass::AssignLayout<descriptor::layout::DenseTensorViewLayout>>();
-    string common_function_string;
+    auto allocator = std::make_shared<runtime::gpu::GPUAllocator>(
+        m_shared_context->m_primitive_emitter->get_memory_allocator());
+
+    m_pass_manager.register_pass<ngraph::pass::ResultCopyElimination>();
+
+    m_pass_manager
+        .register_pass<ngraph::pass::AssignLayout<descriptor::layout::DenseTensorViewLayout>>();
+
+    m_pass_manager.register_pass<ngraph::pass::Liveness>();
+
+    m_pass_manager.register_pass<ngraph::pass::MemoryLayout>(64);
```

Comment:
We should probably have the magic number 64 defined as a constant or something

I just found `runtime::gpu::GPUMemoryManager::alignment`

Reply:
Done, added a static variable for the pool alignment.